### DISCUSSION TITLE
Restrict access to register router

### DISCUSSION THREAD

**Discussion by davidbrochart** [fastapi-users/fastapi-users#1057] [Q&A] [answered]
The register router is not protected in FastAPI-Users, and so anybody can create a new user. But we can imagine a system where only an "admin user" can create new users, and so the register router would only be accessible by him (this admin user would have been registered in the database through another means).
What would be the best way to protect the register router? I was thinking about passing my own `current_user` dependency to [get_register_router](https://github.com/fastapi-users/fastapi-users/blob/1b6b58c0e0040a9efa50d3c8ca4d943cb5f6bf6e/fastapi_users/router/register.py#L10).

**Comment by frankie567** [comment, marked as answer]
The solution I would recommend is to add a router dependency with the check you need. For example, if you want to restrict `/register` to superusers:

```py
current_superuser = fastapi_users.current_user(active=True, superuser=True)

app.include_router(
    fastapi_users.get_register_router(UserRead, UserCreate),
    prefix="/auth",
    tags=["auth"],
    dependencies=[Depends(current_superuser)],
)
```

This way, you don't even need to change the router logic.

**Reply by davidbrochart** [reply]
Thanks a lot!
